src/catalog/function_catalog.cpp

```diff
@@ -0,0 +1,348 @@
+      auto val1 = type::ValueFactory::GetVarcharValue(proname, nullptr);
+      auto val2 = type::ValueFactory::GetNullValueByType(type::Type::INTEGER);
+      auto val3 = type::ValueFactory::GetNullValueByType(type::Type::INTEGER);
+      auto val4 = type::ValueFactory::GetIntegerValue(prolang);
+      auto val5 = type::ValueFactory::GetNullValueByType(type::Type::DECIMAL);
+      auto val6 = type::ValueFactory::GetNullValueByType(type::Type::DECIMAL);
+      auto val7 = type::ValueFactory::GetNullValueByType(type::Type::INTEGER);
+      auto val8 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val9 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val10 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val11 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val12 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val13 = type::ValueFactory::GetNullValueByType(type::Type::BOOLEAN);
+      auto val14 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+      auto val15 = type::ValueFactory::GetIntegerValue(pronargs);
+      auto val16 = type::ValueFactory::GetNullValueByType(type::Type::INTEGER);
+      auto val17 = type::ValueFactory::GetIntegerValue(prorettype);
+
+      std::stringstream os;
+      for(oid_t param_type : proargtypes)
+            os << std::to_string(param_type) << "";
+      
+      auto val18 = type::ValueFactory::GetVarcharValue(os.str(), nullptr);
+    
+    
+      auto val19 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+      auto val20 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+      auto val21 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+     
+      std::stringstream os2;
+      for(auto param_name : proargnames)
+            os2 << param_name  << "";
+      auto val22 = type::ValueFactory::GetVarcharValue(os2.str(), nullptr);
+
+      auto val23 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+      auto val24 = type::ValueFactory::GetVarcharValue(prosrc_bin[0], nullptr);
+
+      auto val25 =  type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+
+      if(prosrc_bin.size() == 1) //only src, no binary
+      {
+        val25 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+      }
+      else { 
+        val25 = type::ValueFactory::GetVarcharValue(prosrc_bin[1], nullptr);
+      }
+     
+      auto val26 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+      auto val27 = type::ValueFactory::GetNullValueByType(type::Type::VARCHAR);
+ 
+
+      tuple->SetValue(0, val0, pool);
+      tuple->SetValue(1, val1, pool);
+      tuple->SetValue(2, val2, pool);
+      tuple->SetValue(3, val3, pool);
+      tuple->SetValue(4, val4, pool);
+      tuple->SetValue(5, val5, pool);
+      tuple->SetValue(6, val6, pool);
+      tuple->SetValue(7, val7, pool);
```

Comment:
Line 287 not sure it there's bug here. got error like this "Message :: INTEGER is not coercable to BOOLEAN" and test just aborts.